src/logic/hooks/useCanTxExecute.tsx

```diff
@@ -1,30 +1,10 @@
-import { useEffect, useState } from 'react'
 import { useSelector } from 'react-redux'
-import { extractSafeAddress } from 'src/routes/routes'
 import { currentSafe } from '../safe/store/selectors'
-import useGetRecommendedNonce from './useGetRecommendedNonce'
 
-export const calculateCanTxExecute = (
-  currentSafeNonce: number,
-  preApprovingOwner: string,
-  threshold: number,
-  txConfirmations: number,
-  recommendedNonce?: number,
-  isExecution?: boolean, // when executing from the TxList
-  manualSafeNonce?: number,
-): boolean => {
-  if (isExecution) return true
-
-  // Single owner
-  if (threshold === 1) {
-    // nonce was changed manually to be executed
-    if (manualSafeNonce) {
-      return manualSafeNonce === currentSafeNonce
-    }
-    // is next tx
-    return recommendedNonce === currentSafeNonce
-  }
+type UseCanTxExecuteType = (preApprovingOwner?: string, txConfirmations?: number) => boolean
 
+const useCanTxExecute: UseCanTxExecuteType = (preApprovingOwner = '', txConfirmations = 0) => {
```

Comment:
Should we also rename this to `canTxExecute` now that it is not a custom hook anymore?

Reply:
Looks like it needs to be a hook because it uses a selector in it.